packages/rulesets/src/asyncapi/functions/utils/getAllMessages.ts

```diff
@@ -0,0 +1,40 @@
+import { isPlainObject } from '@stoplight/json';
+
+import { getAllOperations } from './getAllOperations';
+
+import type { JsonPath } from '@stoplight/types';
+
+type MessageObject = Record<string, unknown>;
+type AsyncAPI = {
+  channels?: Record<string, { subscribe?: Record<string, unknown>; publish?: Record<string, unknown> }>;
+};
+type Result = { path: JsonPath; message: MessageObject };
+
+export function* getAllMessages(asyncapi: AsyncAPI): IterableIterator<Result> {
+  for (const { path, operation } of getAllOperations(asyncapi)) {
```

Comment:
Dunno why, but I thought we had access to a parsed instance and we could use https://github.com/asyncapi/parser-js/blob/58d0505e23fb4b65333dfd25d6b89b0924622367/lib/models/asyncapi.js#L170

But I guess not 😄

Reply:
We don't have it that good 😄